src/Nethermind/Nethermind.JsonRpc.Test/RpcTest.cs

```diff
@@ -63,9 +59,19 @@ private static IJsonRpcService BuildRpcService<T>(T module) where T : class, IRp
         return service;
     }
 
-    public static JsonRpcRequest GetJsonRequest(string method, params string[]? parameters)
```

Comment:
Regarding similar RPC tests fixes in #7547 - I see it's still not in master and actually waiting in #7483.
- Reverting my fixes will break all or most of new tests in this PR
- I can try to cherry-pick #7547, but that will add around 10 unrelated files to changes.

Reply:
#7483 is now on master now

Reply:
I personally liked initial approach better, one with the conditional serialization and custom array construction. While not looking super nice, it accepted both objects and serialized JSON strings as parameters, and latter ones are much easier to pass into tests via `TestCase` attribute.

Reply:
Your point makes sense, specially when dealing with `TestCase` attributes. Feel free to create a PR if you want to revisit this topic